editor/gui/editor_file_dialog.cpp

```diff
@@ -1712,6 +1713,18 @@ bool EditorFileDialog::are_previews_enabled() {
 	return previews_enabled;
 }
 
+void EditorFileDialog::add_side_menu(const String &p_title, Control *p_menu, real_t p_stretch_ratio) {
+	VBoxContainer *side_vbox = memnew(VBoxContainer);
+	side_vbox->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	side_vbox->set_stretch_ratio(p_stretch_ratio);
+	Label *title_label = memnew(Label(p_title));
+	title_label->set_theme_type_variation("HeaderSmall");
+	side_vbox->add_child(title_label);
+	p_menu->set_v_size_flags(Control::SIZE_EXPAND_FILL);
+	side_vbox->add_child(p_menu);
+	body_hbox->add_child(side_vbox);
```

Comment:
```suggestion
	VBoxContainer *side_vbox = memnew(VBoxContainer);
	side_vbox->set_h_size_flags(Control::SIZE_EXPAND_FILL);
	side_vbox->set_stretch_ratio(p_stretch_ratio);
	body_hbox->add_child(side_vbox);

	Label *title_label = memnew(Label(p_title));
	title_label->set_theme_type_variation("HeaderSmall");
	side_vbox->add_child(title_label);
	
	p_menu->set_v_size_flags(Control::SIZE_EXPAND_FILL);
	side_vbox->add_child(p_menu);
```

Something along those lines, once other changes are done.